gtda/homology/tests/test_simplicial.py

```diff
@@ -123,6 +116,109 @@ def test_vrp_fit_transform_plot(X, metric, hom_dims):
         )
 
 
+def test_wrp_params():
+    metric = 'not_defined'
+    wrp = WeightedRipsPersistence(metric=metric)
+
+    with pytest.raises(ValueError):
+        wrp.fit_transform(X_pc)
+
+
+def test_wrp_not_fitted():
+    wrp = WeightedRipsPersistence()
+
+    with pytest.raises(NotFittedError):
+        wrp.transform(X_pc)
+
+
+def test_wrp_notimplemented_string_weights():
+    wrp = WeightedRipsPersistence(weights="foo")
+
+    with pytest.raises(ValueError, match="'foo' passed for `weights` but the "
+                                         "only allowed string is 'DTM'"):
+        wrp.fit(X_pc)
+
+
+def test_wrp_notimplemented_p():
+    wrp = WeightedRipsPersistence(weight_params={'p': 1.2})
+
+    with pytest.raises(ValueError):
+        wrp.fit(X_pc)
+
+
+@pytest.mark.parametrize('X, metric', [(X_pc, 'euclidean'),
+                                       (X_pc_list, 'euclidean'),
+                                       (X_pc_sparse, 'euclidean'),
+                                       (X_dist, 'precomputed'),
+                                       (X_dist_list, 'precomputed'),
+                                       (X_dist_sparse, 'precomputed')])
+@pytest.mark.parametrize('weight_params', [{'p': 1}, {'p': 2}, {'p': np.inf}])
+@pytest.mark.parametrize('collapse_edges', [True, False])
+@pytest.mark.parametrize('max_edge_weight', [np.inf, 0.8])
+@pytest.mark.parametrize('infinity_values', [10, 30])
```

Comment:
So, for that particular point cloud, all of these parameter values give the same result?

Reply:
I am testing that zero weights give the same as vanilla VR, so it would be true for any point cloud if I use the same expected results as for vanilla VR,